ydb/core/kqp/provider/yql_kikimr_exec.cpp

```diff
@@ -1628,22 +1636,69 @@ class TKiSinkCallableExecutionTransformer : public TAsyncCallbackTransformer<TKi
                                     return SyncError();
                                 }
                             }
-                        } else {
-                            ctx.AddError(TIssue(ctx.GetPosition(nameNode.Pos()),
-                                TStringBuilder() << "Unknown add index setting: " << name));
+                        } else if (name == "indexSettings") {
+                            YQL_ENSURE(add_index->type_case() == Ydb::Table::TableIndex::kGlobalVectorKmeansTreeIndex);
+                            auto& protoVectorSettings = *add_index->mutable_global_vector_kmeans_tree_index()->mutable_vector_settings();
+                            auto indexSettings = columnTuple.Item(1).Cast<TCoAtomList>();
+                            YQL_ENSURE(indexSettings.Maybe<TCoNameValueTupleList>());
+                            for (const auto& vectorSetting : indexSettings.Cast<TCoNameValueTupleList>()) {
+                                YQL_ENSURE(vectorSetting.Value().Maybe<TCoAtom>());
+                                if (vectorSetting.Name().Value() == "distance") {
+                                    auto parseEnum = [] (const TString distance) {
```

Comment:
```suggestion
                                    auto parseEnum = [] (const auto& distance) {
```

This comment also for other settings